File: src/qaly.java
import java.util.Scanner;

public class qaly {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        int inputs;
        double total = 0, quality, years;

        inputs = in.nextInt();
        for (int i = 0; i < inputs; i++) {
            quality = in.nextDouble();
            years = in.nextDouble();
            total += qaly( quality, years );
        }

        System.out.printf( "%.3f", total );
    }

    public static double qaly( double quality, double years) {
        return quality * years;
    }
}

File: go/qaly.go
package main

import (
	"fmt"
)

func main() {
	var inputs int
	var total, quality, years float64

	fmt.Scanln( &inputs )
	for i := 0; i < inputs; i++  {
		fmt.Scanln( &quality, &years )
		total += qaly( quality, years )
	}

	fmt.Printf( "%.3f", total )
}

func qaly(quality float64, years float64) float64 {
	return quality * years
}